.pre-commit-config.yaml

```
@@ -19,18 +23,119 @@ repos:
       - id: black
         files: (\.py$|SConstruct|SCsub)
         types_or: [text]
-        exclude: .*thirdparty.*
         args:
           - --line-length=120
 
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.971
+    hooks:
+      - id: mypy
+        files: \.py$
+        types_or: [text]
+        args: [--config-file=./misc/scripts/mypy.ini]
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+      - id: codespell
+        types_or: [text]
+        exclude: |
+          (?x)^(
+            .*\.desktop$|
+            .*\.gitignore$|
+            .*\.po$|
+            .*\.pot$|
+            .*\.rc$|
+            \.mailmap$|
+            AUTHORS.md$|
+            COPYRIGHT.txt$|
+            DONORS.md$|
+            core/input/gamecontrollerdb.txt$|
+            core/string/locales.h$|
+            editor/project_converter_3_to_4.cpp$|
+            platform/android/java/lib/src/com/.*|
+            platform/web/package-lock.json$
+          )
+        args:
+          - --enable-colors
+          - --write-changes
+          - --check-hidden
+          - --quiet-level
+          - '3'
+          - --ignore-words-list
+          - aesthetic,aesthetics,breaked,cancelled,colour,curvelinear,doubleclick,expct,findn,gird,hel,inout,lod,mis,nd,numer,ot,requestor,te,thirdparty,vai
+          - --builtin
+          - clear,rare,en-GB_to_en-US
+
+  ### Requires Docker; look into alternative implementation.
+  # - repo: https://github.com/comkieffer/pre-commit-xmllint.git
+  #   rev: 1.0.0
+  #   hooks:
+  #     - id: xmllint
+  #       language: docker
+  #       types_or: [text]
+  #       files: ^(doc/classes|.*/doc_classes)/.*\.xml$
+  #       args: [--schema, doc/class.xsd]
+
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: v8.46.0
+    hooks:
+    - id: eslint
+      name: eslint-engine
+      files: ^(platform/web/js/engine/|js/jsdoc2rst/).*\.js$
+      args: [--fix, --no-eslintrc, --config, platform/web/.eslintrc.engine.js]
+      additional_dependencies:
+      - eslint@8.46.0
+      - eslint-config-airbnb-base@15.0.0
+      - eslint-plugin-import@2.28.0
```

Shouldn't these be indented? Same below.
```suggestion
      additional_dependencies:
        - eslint@8.46.0
        - eslint-config-airbnb-base@15.0.0
        - eslint-plugin-import@2.28.0
```